hyperspace/cosmos/src/provider.rs

```diff
@@ -0,0 +1,368 @@
+		todo!()
+	}
+
+	async fn query_packet_commitments(
+		&self,
+		at: Height,
+		channel_id: ChannelId,
+		port_id: PortId,
+	) -> Result<Vec<u64>, Self::Error> {
+		todo!()
+	}
+
+	async fn query_packet_acknowledgements(
+		&self,
+		at: Height,
+		channel_id: ChannelId,
+		port_id: PortId,
+	) -> Result<Vec<u64>, Self::Error> {
+		todo!()
+	}
+
+	async fn query_unreceived_packets(
+		&self,
+		at: Height,
+		channel_id: ChannelId,
+		port_id: PortId,
+		seqs: Vec<u64>,
+	) -> Result<Vec<u64>, Self::Error> {
+		todo!()
+	}
+
+	async fn query_unreceived_acknowledgements(
+		&self,
+		at: Height,
+		channel_id: ChannelId,
+		port_id: PortId,
+		seqs: Vec<u64>,
+	) -> Result<Vec<u64>, Self::Error> {
+		todo!()
+	}
+
+	fn channel_whitelist(&self) -> Vec<(ChannelId, PortId)> {
+		todo!()
+	}
+
+	async fn query_connection_channels(
+		&self,
+		at: Height,
+		connection_id: &ConnectionId,
+	) -> Result<QueryChannelsResponse, Self::Error> {
+		todo!()
+	}
+
+	async fn query_send_packets(
+		&self,
+		channel_id: ChannelId,
+		port_id: PortId,
+		seqs: Vec<u64>,
+	) -> Result<Vec<PacketInfo>, Self::Error> {
+		todo!()
+	}
+
+	async fn query_recv_packets(
+		&self,
+		channel_id: ChannelId,
+		port_id: PortId,
+		seqs: Vec<u64>,
+	) -> Result<Vec<PacketInfo>, Self::Error> {
+		todo!()
+	}
+
+	fn expected_block_time(&self) -> Duration {
+		todo!()
+	}
+
+	async fn query_client_update_time_and_height(
+		&self,
+		client_id: ClientId,
+		client_height: Height,
+	) -> Result<(Height, Timestamp), Self::Error> {
+		todo!()
+	}
+
+	async fn query_host_consensus_state_proof(
+		&self,
+		height: Height,
+	) -> Result<Option<Vec<u8>>, Self::Error> {
+		todo!()
+	}
+
+	async fn query_ibc_balance(&self) -> Result<Vec<PrefixedCoin>, Self::Error> {
+		todo!()
+	}
+
+	fn connection_prefix(&self) -> CommitmentPrefix {
+		todo!()
+	}
+
+	fn client_id(&self) -> ClientId {
+		self.client_id()
+	}
+
+	fn client_type(&self) -> ClientType {
+		todo!()
+	}
+
+	fn connection_id(&self) -> ConnectionId {
+		todo!()
+	}
+
+	async fn query_timestamp_at(&self, block_number: u64) -> Result<u64, Self::Error> {
+		todo!()
+	}
+
+	async fn query_clients(&self) -> Result<Vec<ClientId>, Self::Error> {
+		todo!()
+	}
+
+	async fn query_channels(&self) -> Result<Vec<(ChannelId, PortId)>, Self::Error> {
+		todo!()
+	}
+
+	async fn query_connection_using_client(
+		&self,
+		height: u32,
+		client_id: String,
+	) -> Result<Vec<IdentifiedConnection>, Self::Error> {
+		todo!()
+	}
+
+	fn is_update_required(
+		&self,
+		latest_height: u64,
+		latest_client_height_on_counterparty: u64,
+	) -> bool {
+		todo!()
+	}
+
+	async fn initialize_client_state(
+		&self,
+	) -> Result<(AnyClientState, AnyConsensusState), Self::Error> {
+		todo!()
+	}
+
+	async fn query_client_id_from_tx_hash(
+		&self,
+		tx_hash: sp_core::H256,
+		_block_hash: Option<sp_core::H256>,
+	) -> Result<ClientId, Self::Error> {
+		
+		const WAIT_BACKOFF: Duration = Duration::from_millis(300);
```

Comment:
Instead of having to wait for the transaction to included in a block here, it would be better for the function used to submit transactions do the waiting before returning the transaction hash, so in here we just query once and expect
a valid result.

Reply:
Done [6c8d3dc](https://github.com/ComposableFi/centauri/pull/114/commits/6c8d3dc20bc2b8f1e741435d68c9eea78e7a28c9)